README.md

```diff
@@ -38,33 +40,82 @@ On your `_ViewImports.cshtml` add the `using` and TagHelper entries:
 On your .cshtml add a `BECanvas` and make sure you set the `ref` to a field on your component:
 
 ```c#
-@page "/"
-@inherits IndexComponent
-
-<h1>Canvas demo!!!</h1>
-
 <BECanvas ref="@_canvasReference"></BECanvas>
 ```
 
-On your component C# code (regardless if inline on .cshtml or in a .cs file), from a `BECanvasComponent` reference, create a `Canvas2dContext`, and then use the context methods to draw on the canvas: 
+### 2D
+
+On your component C# code (regardless if inline on .razor or in a .cs file), from a `BECanvasComponent` reference, create a `Canvas2DContext`, and then use the context methods to draw on the canvas:
 
 ```c#
-private Canvas2dContext _context;
+private Canvas2DContext _context;
 
 protected BECanvasComponent _canvasReference;
 
-protected override void OnAfterRender()
+protected override async Task OnAfterRenderAsync()
 {
-    this._context = this._canvasReference.CreateCanvas2d();
-    this._context.FillStyle = "green";
+    this._context = await this._canvasReference.CreateCanvas2DAsync();
+    await this._context.SetFillStyleAsync("green");
+
+    await this._context.FillRectAsync(10, 100, 100, 100);
+
+    await this._context.SetFontAsync("48px serif");
+    await this._context.StrokeTextAsync("Hello Blazor!!!", 10, 100);
+}
+```
+
+**NOTE** You cannot call `CreateCanvas2DAsync` in `OnInitAsync`, because the underlying `&lt;canvas&gt;` element is not yet present in the generated markup.
+
+### WebGL
 
-    this._context.FillRect(10, 100, 100, 100);
+On your component C# code (regardless if inline on .razor or in a .cs file), from a `BECanvasComponent` reference, create a `WebGLContext`, and then use the context methods to draw on the canvas:
+
+```c#
+private WebGLContext _context;
+
+protected BECanvasComponent _canvasReference;
 
-    this._context.Font = "48px serif";
-    this._context.StrokeText("Hello Blazor!!!", 10, 100);
+protected override async Task OnAfterRenderAsync()
+{
+    this._context = await this._canvasReference.CreateWebGLAsync(new WebGLContextAttributes
+    {
+        PowerPreference = WebGLContextAttributes.POWER_PREFERENCE_HIGH_PERFORMANCE
```

Comment:
you should use the parameterless create method here. I added this for testing purposes

Reply:
Thanks for this. I've updated the PR.